packages/react-hooks/src/useAsyncInterval.ts

```diff
@@ -64,6 +69,7 @@ export function useAsyncInterval(
     setTimeoutRef.current = window.setTimeout(tick, targetIntervalMs);
 
     return () => {
+      isCancelledRef.current = true;
```

Comment:
This should be in it's own `useEffect`, with no dependencies so it's only called on `unmount`. This will "cancel" it anytime there's a change to `callback` or to `targetIntervalMs`, which would not be correct.

Reply:
good catch. I split out a separate hook